src/app/context/context.component.ts

```diff
@@ -81,18 +81,21 @@ export class ContextComponent implements OnInit {
   constructor(
     public router: Router,
     private route: ActivatedRoute,
-    public dialog: MatDialog) {
+    public dialog: MatDialog,
+    private fetchActivityService: FetchActivityService) {
   }
 
   ngOnInit(): void {
-    // get the id from the current route
-    const routeParams = this.route.snapshot.paramMap;
-    const cardIdFromRoute = Number(routeParams.get('id'));
 
-    // find the deck for id we got from the route
-    this.activityData = allContextDecksCollection.find((array: { id: number; }) => Number(array.id) === cardIdFromRoute);
+    const routeSegment = this.route.snapshot.url[0].path;
+    const activityId = Number(this.route.snapshot.url[1].path);
+
+    this.fetchActivityService.fetchActivity(routeSegment, activityId)
```

Comment:
activityService.fetch @leventemo

Reply:
Renaming of the service file & service fns done.